numpy/core/_methods.py

```diff
@@ -140,13 +141,18 @@ def _clip(a, min=None, max=None, out=None, *, casting=None, **kwargs):
         return _clip_dep_invoke_with_casting(
             um.clip, a, min, max, out=out, casting=casting, **kwargs)
 
-def _mean(a, axis=None, dtype=None, out=None, keepdims=False):
+def _mean(a, axis=None, dtype=None, out=None, keepdims=False, where=True):
     arr = asanyarray(a)
 
     is_float16_result = False
-    rcount = _count_reduce_items(arr, axis)
-    # Make this warning show up first
-    if rcount == 0:
+
+    if isinstance(where, bool):
+        rcount = _count_reduce_items(arr, axis)
+    else:
+        logic_mask = where * nu.ones_like(arr, dtype=nt.bool_)
+        rcount = umr_sum(logic_mask, axis, nt.intp, None, keepdims)
```

Comment:
If you moved these 5 lines into ` _count_reduce_items`, then I think you'd get `where` for free in `np.var` too.

Reply:
Should be possible, although I still have to figure out how to abstract this part in `_var`:
```
    # Compute degrees of freedom and make sure it is not negative.
    rcount = max([rcount - ddof, 0])
```

I will add a separate to the same PR later.

Reply:
`np.maximum(rcount - ddof, 0)`

Reply:
Thank you, I figured it out and thought 'Nice, this was easy' but then I was fiddling a little with broadcasting (`arrmean` needs to have fixed dimensions but `rcount` can cause wrong broadcasting behavior depending on where it is fixed-dim or not).